### PR TITLE
Update the P1NC element implementation.

### PR DESCRIPTION
**Pull request by bangerth**
Specifically, this updates the class documentation for language
and clarity. It also tightens up the code in a number of places.

This is an update to the just committed #2851.

### CODE REVIEW

File: include/deal.II/fe/fe_p1nc.h [hunk 1]
```diff
@@ -29,25 +29,26 @@ DEAL_II_NAMESPACE_OPEN
...
+ * Thus, each function in the nonconforming element space can be
+ * discontinuous, and consequently not included in $H^1_0$, just like
+ * the basis functions in Discontinuous Galerkin (DG) finite element
+ * spaces. On the other hand, basis functions in DG spaces are
+ * completely discontinuous across edges without any relation between
+ * the values from both sides.  This is a reason why usual weak
+ * formulations for DG schemes contain additional penalty terms for
+ * jump across edges to control discontinuity.  However nonconforming
+ * elements usually do not need additional terms in their weak
+ * formulations due to the continuity in integral on edges.
```
Comment (tjhei) [inline include/deal.II/fe/fe_p1nc.h]: "continuity in integral" -> "continuity of the average" ?

File: include/deal.II/fe/fe_p1nc.h [hunk 2]
```diff
@@ -62,29 +63,29 @@ DEAL_II_NAMESPACE_OPEN
  * However, these 4 functionals are not linearly independent
  * because a linear function on 2D is uniquely determined by only 3 independent values.
  * A simple observation reads that any linear function on a quadrilateral should satisfies the 'dice rule':
```
Comment (tjhei) [inline include/deal.II/fe/fe_p1nc.h]: "satisfy"

File: include/deal.II/fe/fe_p1nc.h [hunk 3]
```diff
@@ -102,13 +103,17 @@ DEAL_II_NAMESPACE_OPEN
  *  @endverbatim
  *
  * For each vertex $v_j$ of given cell, there are two edges of which $v_j$ is one of end points.
- * Consider a linear function such that 0.5 value at two midpoints of such edges,
- * and 0.0 at two midpoints of other edges.
+ * Consider a linear function such that it has value 0.5 at midpoints at the midpoints of two adjacent edges,
```
Comment (tjhei) [inline include/deal.II/fe/fe_p1nc.h]: remove "at midpoints"

File: include/deal.II/fe/fe_p1nc.h [hunk 4]
```diff
@@ -102,13 +103,17 @@ DEAL_II_NAMESPACE_OPEN
...
+ * and 0.0 at two midpoints of the other edges.
```
Comment (tjhei) [inline include/deal.II/fe/fe_p1nc.h]: "at the two"?

File: include/deal.II/fe/fe_p1nc.h [hunk 5]
```diff
@@ -173,61 +178,63 @@ DEAL_II_NAMESPACE_OPEN
  *
  * </ul>
  *
- * The local DOFs are defined by the coefficients of the shape functions associated vertices, respectively.
- * Although these 4 local DOFs are not linearly independent within a single cell as well,
+ * The local DOFs are defined by the coefficients of the shape functions associated with vertices, respectively.
```
Comment (tjhei) [inline include/deal.II/fe/fe_p1nc.h]: we typically spell if "DoFs"

File: source/fe/fe_p1nc.cc [hunk 6]
```diff
@@ -198,10 +194,10 @@ FE_P1NC::get_subface_data (const UpdateFlags update_flags,
 
 void
 FE_P1NC::fill_fe_values (const Triangulation<2,2>::cell_iterator           &cell,
-                         const CellSimilarity::Similarity                   cell_similarity,
-                         const Quadrature<2>                               &quadrature,
-                         const Mapping<2,2>                                &mapping,
-                         const Mapping<2,2>::InternalDataBase              &mapping_internal,
+                         const CellSimilarity::Similarity                   ,
```
Comment (tjhei) [inline source/fe/fe_p1nc.cc]: is this really what astyle wants? funny.

**Comment by bangerth** [comment]
All comments addressed.

File: doc/doxygen/scripts/make_gallery.pl [hunk 7]
```diff
@@ -86,7 +86,7 @@
 my @picture_files;
 foreach my $file (@src_files)
 { 
-    if ($file =~ /.*\.(png|jpg|gif|svg)/)
+    if ($file =~ /.*\.(png|jpg|gif|svg|mp4)/)
```
Comment (tjhei) [inline doc/doxygen/scripts/make_gallery.pl]: You added some unrelated changes!

**Comment by tjhei** [comment]
Ready to be merged if you remove the unrelated changes.

**Comment by bangerth** [comment]
Ah, cr*p. I should have learned long ago to not use `-a` :-(

Fixed.
